### PR TITLE
Aggregate all validation needed into "make validate"

### PR DESCRIPTION
**Pull request by tylerslaton**
There are now two levels of validation:
- "make validate-code" - Called by test action
- "make validate-docs" - Called by docs action

These are both aggregated by "make validate" to make
preparing a commit for push easier. It will perform
all the necassary checks.

Additionally:
- Delete "make validate-ci"
- Update generate.go to use go run for mockgen
- Delete "make mocks" in favor of running "make generate"

### Checklist
- [X] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [X] The title of this PR ends with a link to the main issue being address in paranthesis.
- [X] All relevant issues are referenced in the PR description.
- [X] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [X] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [X] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,33 +17,32 @@ setup-ci-image:
 	docker build -t acorn:v-ci .
 	docker save acorn:v-ci | docker exec -i $$(docker ps | grep k3s | awk '{print $$1}') ctr --address /run/k3s/containerd/containerd.sock images import -
 
-validate:
+GOLANGCI_LINT_VERSION := v1.51.1
+lint:
+	if ! command -v golangci-lint &> /dev/null; then \
+  		echo "Could not find golangci-lint, installing version $(GOLANGLINT_CI_VERSION)."; \
+		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin $(GOLANGCI_LINT_VERSION); \
+	fi
```
Comment (thedadams) [inline Makefile:25]: So the version can be easily overwritten by an environment variable if need be.
```suggestion
lint:
	if ! command -v golangci-lint &> /dev/null; then \
  		echo "Could not find golangci-lint, installing version ${GOLANGLINT_CI_VERSION:-v1.51.1}."; \
		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin ${GOLANGCI_LINT_VERSION:-v1.51.1}; \
	fi
```

File: Makefile [hunk 2]
```diff
@@ -14,33 +17,32 @@ setup-ci-image:
...
-	;fi
+	;fi 
 
 test:
 	go test $(TEST_FLAGS) ./...
 
 goreleaser:
 	goreleaser build --snapshot --single-target --rm-dist
 
-setup-ci-env:
```
Comment (thedadams) [inline Makefile:37]: I like the idea of keeping this and having `lint` call this. Reason being that we may need to add more tools like this in the future.

Thoughts?

File: Makefile (hunk 1 above)
Comment (tylerslaton) [reply Makefile]: I wasn't able to get this specific change working locally. However, I was able to implement the intent by changing the variable declaration to be a `?=`. Thoughts?

**Review by thedadams — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -14,13 +17,21 @@ setup-ci-image:
 	docker build -t acorn:v-ci .
 	docker save acorn:v-ci | docker exec -i $$(docker ps | grep k3s | awk '{print $$1}') ctr --address /run/k3s/containerd/containerd.sock images import -
 
-validate:
+GOLANGCI_LINT_VERSION ?= v1.51.1
+setup-env: 
+	if ! command -v golangci-lint &> /dev/null; then \
+  		echo "Could not find golangci-lint, installing version $(GOLANGLINT_CI_VERSION)."; \
```
Comment (g-linville) [inline Makefile:23]: Nit:

```suggestion
  		echo "Could not find golangci-lint, installing version $(GOLANGCI_LINT_VERSION)."; \
```

Comment (tylerslaton) [reply Makefile:23]: Oh, nice catch. Thanks!

**Review by g-linville — APPROVED** [review]

**Pull request merged**